beeai/issue_fetcher/jira_issue_fetcher.py

```diff
@@ -0,0 +1,381 @@
+        """
+        Search for issues using the configured query with pagination
+        """
+        logger.info(f"Starting issue search with query: {self.query}")
+        
+        all_issues = []
+        start_at = 0
+        total_issues = None
+        
+        fields = [
+            "key",        # Issue key (e.g., RHEL-12345)
+            "labels",     # Issue labels 
+        ]
+        
+        while True:
+            await self._rate_limit()
+            
+            # Use POST with JSON payload instead of GET with params to handle long queries
+            json_payload = {
+                "jql": self.query,
+                "startAt": start_at,
+                "maxResults": self.max_results_per_page,
+                "fields": fields
+            }
+            
+            logger.info(f"Fetching issues: startAt={start_at}, maxResults={self.max_results_per_page}")
+            
+            try:
+                url = urljoin(self.jira_url, "rest/api/2/search")
+                response_data = self._make_request_with_retries(url, json_data=json_payload)
+                
+                issues = response_data.get("issues", [])
+                all_issues.extend(issues)
+                
+                if total_issues is None:
+                    total_issues = response_data.get("total", 0)
+                    logger.info(f"Total issues found: {total_issues}")
+                
+                logger.info(f"Retrieved {len(issues)} issues (total so far: {len(all_issues)}/{total_issues})")
+                
+                if len(all_issues) >= total_issues or len(issues) == 0:
+                    break
+                
+                start_at += self.max_results_per_page
+                
+            except Exception as e:
+                logger.error(f"Error fetching issues: {e}")
+                raise
+        
+        logger.info(f"Successfully retrieved {len(all_issues)} issues")
+        return all_issues
+    
+    @asynccontextmanager
+    async def redis_client(self) -> AsyncGenerator[redis.Redis, None]:
+        client = redis.Redis.from_url(self.redis_url)
+        try:
+            await client.ping()
+            logger.info("Connected to Redis")
+            yield client
+        finally:
+            await client.aclose()
+            logger.info("Disconnected from Redis")
+    
+    async def _get_existing_issue_keys(self, redis_conn) -> set:
+        """
+        Get all existing issue keys from all Redis queues to avoid duplicates
+        """
+        try:
+            # All Redis queues and lists to check
+            queue_names = [
+                "triage_queue",
+                "rebase_queue", 
+                "backport_queue",
+                "clarification_needed_queue",
+                "error_list",
+                "no_action_list",
+                "completed_rebase_list",
+                "completed_backport_list"
+            ]
```

Comment:
shouldn't we be checking here only the first 3? For the other ones, we want to be able to retry those, or am I missing something?

Reply:
but we are checking only the Jira issue key here, while there can be e.g. a new comment or a link, for that case, we would still skip the reentering based on the key, or?

Reply:
When I answered your question I was thinking at the labels, now I see we are talking about the redis queues. Probably I shouldn't rely at all on the redis queues but only on the labels. However at the moment the labels are not implemented, so this is the only way I have right now.

And I really need to add some more unit tests. I am working on them I hope I will finish them soon.

Reply:
> but we are checking only the Jira issue key here, while there can be e.g. a new comment or a link, for that case, we would still skip the reentering based on the key, or?

Now the `retry_needed` label should work, it removed the already added key from those skipped.

But as I said, I would like to test all this logic further with unit tests.

Reply:
I would say that once we have the labels implemented for every jotnar workflow action I can remove checking the redis queues.
And the `jotnar_retry_needed` label should be requeued, thanks to [this](https://github.com/packit/ai-workflows/blob/36e884ce8de3bb510f3088fc0ad4709c8094921d/beeai/issue_fetcher/jira_issue_fetcher.py#L295-L297)